pkg/cmd/image/list.go

```diff
@@ -128,6 +130,46 @@ type imagePrintable struct {
 
 func printImages(ctx context.Context, client *containerd.Client, imageList []images.Image, options *types.ImageListOptions) error {
 	w := options.Stdout
+	var ImageList []images.Image
+	/*
+		the same imageId under k8s.io is showing multiple results: repo:tag, repo:digest, configID.
+		We expect to display only repo:tag, consistent with other namespaces and CRI
+		e.g.
+		nerdctl -n k8s.io images
+		REPOSITORY    TAG       IMAGE ID        CREATED        PLATFORM       SIZE         BLOB SIZE
+		centos        7         be65f488b776    3 hours ago    linux/amd64    211.5 MiB    72.6 MiB
+		centos        <none>    be65f488b776    3 hours ago    linux/amd64    211.5 MiB    72.6 MiB
+		<none>        <none>    be65f488b776    3 hours ago    linux/amd64    211.5 MiB    72.6 MiB
+		expect:
+		nerdctl --kube-hide-dupe -n k8s.io images
+		REPOSITORY    TAG       IMAGE ID        CREATED        PLATFORM       SIZE         BLOB SIZE
+		centos        7         be65f488b776    3 hours ago    linux/amd64    211.5 MiB    72.6 MiB
+	*/
+	if options.GOptions.KubeHideDupe && options.GOptions.Namespace == "k8s.io" {
+		imageDigest := make(map[digest.Digest]bool)
+		var ImageNoTag []images.Image
+		for _, img := range imageList {
+			parsed, err := reference.ParseAnyReference(img.Name)
+			if err != nil {
+				continue
+			}
+			if _, ok := parsed.(reference.Tagged); !ok {
+				ImageNoTag = append(ImageNoTag, img)
+				continue
+			}
+			ImageList = append(ImageList, img)
+			imageDigest[img.Target.Digest] = true
+		}
+		//Ensure that dangling images without a repo:tag are displayed correctly.
+		for _, ima := range ImageNoTag {
```

Comment:
https://github.com/containerd/nerdctl/pull/3744#discussion_r1894316679
Here, I need to obtain the interface corresponding to the image name, and then use interface.type to distinguish whether it's a standard tag( reference.Tagged) or a non-standard tag().
I simply expected standard implementation, but `referenceutil.Parse` works too.